kornia/filters/kernels.py

```diff
@@ -13,15 +15,20 @@ def normalize_kernel2d(input: torch.Tensor) -> torch.Tensor:
     return input / (norm.unsqueeze(-1).unsqueeze(-1))
 
 
-def gaussian(window_size: int, sigma: float) -> torch.Tensor:
-    device, dtype = None, None
-    if isinstance(sigma, torch.Tensor):
+def gaussian(window_size: int, sigma: Union[float, torch.Tensor]) -> torch.Tensor:
```

Comment:
@edgarriba the problem is located where `sigma` is generated, if I put the line of code in the function `apply_transform`
```
 sigma = torch.tensor([[2.0085, 2.0085], [2.0794, 2.0794], [1.6040, 1.6040]], device=input.device, dtype=input.dtype)
```

instead of 
```
 sigma = params["sigma"].to(device=input.device, dtype=input.dtype).unsqueeze(-1).expand(-1, 2)
 ```

it works.

Reply:
i see -- I'd suggest to isolate in a separated PR to upgrade that function to use Tensor